src/main/java/org/mskcc/cbio/portal/scripts/ImportTabDelimData.java

```diff
@@ -277,28 +294,71 @@ public void importData(int numLines) throws IOException, DaoException {
+                if (FileUtil.isInfoLine(line)) {
+                    String[] rowParts = line.split("\t", -1);
+
+                    if (rowParts.length > headerColumns && line.split("\t").length > headerColumns) {
+                        ProgressMonitor.logWarning("Ignoring line with more fields (" + rowParts.length
+                                + ") than specified in the headers(" + headerColumns + "): \n" + rowParts[0]);
+                    } else if (rowParts.length < headerColumns) {
+                        ProgressMonitor.logWarning("Ignoring line with less fields (" + rowParts.length
+                                + ") than specified in the headers(" + headerColumns + "): \n" + rowParts[0]);
+                    } else {
+                        String sampleValues[] = ArrayUtils.subarray(rowParts, sampleStartIndex, rowParts.length > headerColumns ? headerColumns : rowParts.length);
+
+                        // trim whitespace from values
+                        sampleValues = Stream.of(sampleValues).map(String::trim).toArray(String[]::new);
+                        sampleValues = filterOutNormalValues(filteredSampleIndices, sampleValues);
+
+                        // either parse line as geneset or gene for importing into 'genetic_alteration' table
+                        if (isGsvaProfile) {
+                            String genesetId = rowParts[genesetIdIndex];
+                            recordAdded = saveGenesetLine(sampleValues, genesetId);
+                        } else if (isGenericAssayProfile) {
+                            String genericAssayId = rowParts[genericAssayIdIndex];
+                            recordAdded = saveGenericAssayLine(sampleValues, genericAssayId, genericAssayStableIdToEntityIdMap);
+                        } else {
+                            String geneSymbol = null;
+                            if (hugoSymbolIndex != -1) {
+                                geneSymbol = rowParts[hugoSymbolIndex];
+                            }
+                            if (rppaGeneRefIndex != -1) {
+                                geneSymbol = rowParts[rppaGeneRefIndex];
+                            }
+                            if (geneSymbol != null && geneSymbol.isEmpty()) {
+                                geneSymbol = null;
+                            }
+                            //get entrez
+                            String entrez = null;
+                            if (entrezGeneIdIndex != -1) {
+                                entrez = rowParts[entrezGeneIdIndex];
+                            }
+                            if (entrez != null && entrez.isEmpty()) {
+                                entrez = null;
+                            }
+                            if (entrez != null && !EntrezValidator.isaValidEntrezId(entrez)) {
+                                ProgressMonitor.logWarning("Ignoring line with invalid Entrez_Id " + entrez);
+                            } else {
```

Comment:
maybe move the entrez parsing to a separate function?

Reply:
I could not think of a function that wouldn't look ugly (by having too many arguments )